### DISCUSSION TITLE
File upload

### DISCUSSION THREAD

**Discussion by Andybonner300** [filamentphp/filament#12806] [Help]
### Package

Form builder

### Package Version

3.2

### How can we help you?

When i upload a file it saves it to the folder but doesn't show in my resource page. 

my current code is 
Forms\Components\Section::make('File Upload')
                ->schema([
                    FileUpload::make('attachment')
                    ->preserveFilenames()
                    ->disk('local')
                    ->directory('form-attachments')
                    ->multiple()
                    ->openable()
                    ->downloadable()
                    ->visibility("public")
                ]),

it was 
Forms\Components\Section::make('File Upload')
                ->schema([
                    FileUpload::make('attachment')
                    ->preserveFilenames()
                    ->disk('local')
                    ->directory('form-attachments')
                    ->multiple()
                    ->openable()
                    ->downloadable()
                ]),
                        
                        ])->compact(),
            ]),
            Forms\Components\Section::make('View Uploaded File')
            ->visible(function ($record) {
                // Only show this section if a file is uploaded
                return $record->attachment !== null;
            })
            ->schema([
                // Display a link to view the uploaded file
                Forms\Components\TextInput::make('attachment_link')
                    
                    ->label('Uploaded File')
                    ->default(function ($record) {
                        // Generate the URL to view the uploaded file
                        return Storage::public($record->attachment);
                    }),
                ]),  

neither of these show the file after i have uploaded it. 
File disk is set to local 
FILESYSTEM_DISK=local
 
'disks' => [

        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
            'throw' => false,
        ],

        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
            'throw' => false,
        ],

        's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'throw' => false,
        ],

    ],

    /*
    |--------------------------------------------------------------------------
    | Symbolic Links
    |--------------------------------------------------------------------------
    |
    | Here you may configure the symbolic links that will be created when the
    | `storage:link` Artisan command is executed. The array keys should be
    | the locations of the links and the values should be their targets.
    |
    */

    'links' => [
        public_path('storage') => storage_path('app/public'),
    ],

];

even when i add  ImageColumn::make('attachment')->square(), nothing is showing 

Thank you for your help

**Comment by Tschucki** [comment]
Did you link your storage with `php artisan storage:link`?

**Reply by Andybonner300** [reply]
yeah when i try to do it again it says Documents\sandcrm\public\storage] link already exists.

**Reply by leandrocfe** [reply]
Does it work if you set ->disk('public') ?
